schema/object.go

```diff
@@ -11,6 +11,9 @@ import (
 type Object interface {
 	Type
 	ID() string
+	// HasLooseIDMatching indicates whether the ID must match when validating schema compatibility.
+	// If true, the IDs do not need to match between compared object, but fields must still match.
```

Comment:
Do you think it should be changed?

Reply:
Maybe, but "names are hard". I think that "loose" matching implies there *is* an ID match, but it's somehow fuzzy rather than exact. It's more like "MatchIDs" (or rather, "NoIDMatch"?). However, something akin to "HasLooseTyping" or, yeah, "HasDuckTyping" might not be bad.

Reply:
I renamed it to `IDUnenforced`.

Reply:
Sure, that's an improvement. It still says "between compared object" though. 😆

Reply:
Fixed.